src/main/java/org/alliancegenome/curation_api/jobs/executors/SequenceTargetingReagentExecutor.java

```diff
@@ -29,38 +35,107 @@ public void execLoad(BulkLoadFile bulkLoadFile) {
+					SequenceTargetingReagentIngestFmsDTO.class);
 			bulkLoadFile.setRecordCount(sqtrIngestFmsDTO.getData().size());
 
-			AGRCurationSchemaVersion version = SequenceTargetingReagent.class.getAnnotation(AGRCurationSchemaVersion.class);
+			AGRCurationSchemaVersion version = SequenceTargetingReagent.class
+					.getAnnotation(AGRCurationSchemaVersion.class);
 			bulkLoadFile.setLinkMLSchemaVersion(version.max());
 
-			if (sqtrIngestFmsDTO.getMetaData() != null && StringUtils.isNotBlank(sqtrIngestFmsDTO.getMetaData().getRelease())) {
+			if (sqtrIngestFmsDTO.getMetaData() != null
+					&& StringUtils.isNotBlank(sqtrIngestFmsDTO.getMetaData().getRelease())) {
 				bulkLoadFile.setAllianceMemberReleaseVersion(sqtrIngestFmsDTO.getMetaData().getRelease());
 			}
 
 			BackendBulkDataProvider dataProvider = BackendBulkDataProvider.valueOf(fms.getFmsDataSubType());
 
 			List<Long> sqtrIdsLoaded = new ArrayList<>();
-			List<Long> sqtrIdsBefore = new ArrayList<>();
+			List<Long> sqtrIdsBefore = sqtrService.getIdsByDataProvider(dataProvider.name());
+			List<Long> sqtrGeneAssociationIdsLoaded = new ArrayList<>();
+			List<Long> sqtrGeneAssociationIdsBefore = sqtrService.getIdsByDataProvider(dataProvider.name());
+
 
 			bulkLoadFileDAO.merge(bulkLoadFile);
 
 			BulkLoadFileHistory history = new BulkLoadFileHistory(sqtrIngestFmsDTO.getData().size());
 
-			boolean success = runLoad(sqtrService, history, dataProvider, sqtrIngestFmsDTO.getData(), sqtrIdsLoaded);
+			runLoad(history, dataProvider, sqtrIngestFmsDTO.getData(), sqtrIdsLoaded, sqtrGeneAssociationIdsLoaded);
+
+			runCleanup(sqtrService, history, dataProvider.name(), sqtrIdsBefore, sqtrIdsLoaded, "SQTR", bulkLoadFile.getMd5Sum());
+			runCleanup(sqtrService, history, dataProvider.name(), sqtrGeneAssociationIdsBefore, sqtrGeneAssociationIdsLoaded, "SQTR Gene Associations", bulkLoadFile.getMd5Sum());
 
-			if (success) {
-				runCleanup(sqtrService, history, dataProvider.name(), sqtrIdsBefore, sqtrIdsLoaded, "SQTR", bulkLoadFile.getMd5Sum());
-			}
-			
 			history.finishLoad();
-			
+
 			updateHistory(history);
 		} catch (Exception e) {
 			failLoad(bulkLoadFile, e);
 			e.printStackTrace();
 		}
 	}
 
+	public APIResponse runLoadApi(String dataProviderName, List<SequenceTargetingReagentFmsDTO> sqtrDTOs) {
+		List<Long> sqtrIdsLoaded = new ArrayList<>();
+		List<Long> sqtrGeneAssociationIdsLoaded = new ArrayList<>();
+
+		BulkLoadFileHistory history = new BulkLoadFileHistory(sqtrDTOs.size());
+		BackendBulkDataProvider dataProvider = BackendBulkDataProvider.valueOf(dataProviderName);
+		runLoad(history, dataProvider, sqtrDTOs, sqtrIdsLoaded, sqtrGeneAssociationIdsLoaded);
+		history.finishLoad();
+
+		return new LoadHistoryResponce(history);
+	}
+
+	private void runLoad(BulkLoadFileHistory history, BackendBulkDataProvider dataProvider, List<SequenceTargetingReagentFmsDTO> sqtrs, List<Long> sqtrIdsLoaded, List<Long> sqtrGeneAssociationIdsLoaded) {
```

Comment:
if you can its best to use the generic methods unless these need to do some specific logic